subprojects/core/src/test/groovy/org/gradle/api/internal/project/DefaultProjectTest.groovy

```diff
@@ -990,6 +990,34 @@ def scriptMethod(Closure closure) {
         p.subprojects { assertLifecycleAwareWithReferrer(it, p) }
     }
 
+    def equalsContractForWrappers() {
```

Comment:
🧪 I think we should have integration tests as well, comparing projects from within the different configuration scopes (e.g. `subprojects`, `allprojects`, `rootProject`).

Reply:
Added a test